### PR TITLE
b/172866817: Fix timing issue causing ASan test to fail

### PR DESCRIPTION
**Pull request by nareddyt**
- Add extra sleep time to integration tests due to ASan being slow.

Signed-off-by: Teju Nareddy <nareddyt@google.com>

### CODE REVIEW

**Review by qiwzhang — APPROVED** [review]

**Review by TAOXUY — APPROVED** [review]

**Comment by nareddyt** [comment]
> Should not it be
> 
> time.Duration(tc.wantNumFails * time.Second * 4)
> 
> What does time.Duration(3) mean? 3 seconds?

No, that expression will fail to compile. `tc.wantNumFails` is an int, cannot be multiplied by `time.Second` until it's converted to a duration.

`time.Duration(3)` does not have any unit associated with it, you don't know what unit it is. But when you multiply that by ` * time.Second`, it associates a unit with the duration.

**Review by qiwzhang — APPROVED** [review]

**Comment by nareddyt** [comment]
> Why?
> 
> In this ALWAYS_INIT mode, we should mark it ready when fetching fails.
> 
> In the new mode, START_UP_ASAP, we just don't register, so init_manager will not wait for us.

Ah right, they want a mode that will startup and suspend requests. OK, in that case I'll revert the change here, we can preserve this mode as is.

File: tests/integration_test/iam_imds_data_path_test.go [hunk 1]
```diff
@@ -113,17 +113,19 @@ func TestIamImdsDataPath(t *testing.T) {
...
-				glog.Infof("Sleeping to ensure Envoy is starting")
-				time.Sleep(7 * time.Second)
 			}
 
 			defer s.TearDown(t)
 			if err := s.Setup(tc.confArgs); err != nil {
 				t.Fatalf("fail to setup test env, %v", err)
 			}
 
+			if tc.wantErr != "" {
```
Comment (qiwzhang) [inline tests/integration_test/iam_imds_data_path_test.go:124]: Don't understand why we only need to sleep for wantErr case. Could you add comment on that?

Comment (nareddyt) [reply tests/integration_test/iam_imds_data_path_test.go:125]: Done.

**Review by qiwzhang — APPROVED** [review]

**Pull request merged**